### ISSUE TITLE
Rethink implementation "Accessible during work hours"

### ISSUE THREAD

**Issue by jxpsert** [openstreetmap-polska/openaedmap-frontend#113]
Currently, the icon "Accessible during work hours" is displayed when ``access=customers`` is present. However, the name of the icon implies that it should occur more often. Every AED inside a store would need it, however not every AED in a store is for customers only.
_(imagine that, you're dying outside a store and you don't get saved because the AED is for customers only)_

According to the [OSM wiki](https://wiki.openstreetmap.org/wiki/Key:access), ``access=customers`` means: "Only for customers of the element."

My proposition: any AED that has a value for ``opening_hours`` should get this icon.

**Comment by ttomasz** [comment]
relevant comment about the meaning of `access` tag: https://github.com/openstreetmap-polska/openaedmap-frontend/issues/106#issuecomment-1364603837

**Comment by rbovard** [comment]
Yes, something must be done because it is currently quite confusing.

*Accessible during work hours* is different of *Only for customers*
but *Only for customers* could imply *Accessible during work hours*

**Comment by wzbartczak** [comment]
> 

Do you have any suggestions for combining tags for such markings?

**Comment by jxpsert** [comment]
I'd go for: if ``opening_hours`` has a value, and ``access`` isn't ``no``, ``private`` or anything specifically restrictive, it should be marked on the map as 'Accessible during work hours'.

**Comment by rbovard** [comment]
@jxpsert I do not agree, since the current legend is based on the `access` attribute:

- `access=yes` --> Publicly accessible
- `access=customers` --> Accessible during work hours
- `access=private` --> Accessible with owner's permit
- `access=permissive` --> Publicly available until further notice
- `access=no` --> Unavailable
- No access tag or not matching above` --> Unknown access

It will be more consistant to rename `Accessible during work hours` to its real meaning: `Accessible only for customers`

If you want to handle the `opening_hours`, it should be applied to all legend classes (maybe with an outline, or something to "overwrite" the original symbol). But in this case, the `24/7` should be handled differently as the other values. But I think it's outside the scope of this issue, wich is concerning the translation "Accessible during work hours"

What do you think?

**Comment by jxpsert** [comment]
How often do AEDs ***exclusively*** for customers actually appear? I find it hard to believe that a store, office or other company space would deny people access to an AED if the patient isn't a customer.

For me, it's either checking the existence of ``opening_hours``, or a new value for ``access`` such as ``open``, ``opened``, ``when_open``, etc. That last idea doesn't seem efficient, as it would require a lot of edits on OSM.

**Comment by rbovard** [comment]
> How often do AEDs _**exclusively**_ for customers actually appear? I find it hard to believe that a store, office or other company space would deny people access to an AED if the patient isn't a customer.

If it is in a place where you have to pay, such as a swimming pool or an amusement park?

As I said for the opening_hours, we should discuss it in another issue. There is something to I think.

**Comment by jxpsert** [comment]
> If it is in a place where you have to pay, such as a swimming pool or an amusement park?
Sure, in an amusement park or in a really big fenced off swimming area.

If someone outside a normal swimming pool is dying, nobody will hesitate to take the AED inside the swimming pool outside. Nor should one.

**Comment by wzbartczak** [comment]
@jxpsert, @rbovard , @starsep 

On the occasion of minor analyzes with AED data, I would like to suggest changing the form and using the following values:

* `access=yes`--> Publicly accessible
* `access=customers` --> Accessible during work hours
* `access=private` --> Accessible with owner's permit
* `access=permit` --> Only with the consent of the owner
* `access=permissive` --> Publicly available until further notice
* `access=no` --> Unavailable


and add validation:

if  `access=customers`  plesase add  `opening_hours`
if  `access=yes`  not allowed to add  `opening_hours` and  `opening_hours`  =  24/7 can be added automatically

>If someone outside a normal swimming pool is dying, nobody will hesitate to take the AED inside the swimming pool outside. Nor should one.

Of course, but the access time to the device increases. The user of the map should know this. In addition, many such facilities are not available outside working hours.

**Comment by jxpsert** [comment]
Did you switch private and permit? Surely, if you need the permit of the owner it’s access=permit?
________________________________
Van: Włodzimierz Bartczak ***@***.***>
Verzonden: Thursday, May 4, 2023 10:26:00 AM
Aan: openstreetmap-polska/openaedmap-frontend ***@***.***>
CC: Jasper Platenburg ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [openstreetmap-polska/openaedmap-frontend] Rethink implementation "Accessible during work hours" (Issue #113)


@jxpsert<https://github.com/jxpsert>, @rbovard<https://github.com/rbovard> , @starsep<https://github.com/starsep>

On the occasion of minor analyzes with AED data, I would like to suggest changing the form and using the following values:

  *   access=yes--> Publicly accessible
  *   access=customers --> Accessible during work hours
  *   access=private --> Accessible with owner's permit
  *   access=permit --> Only with the consent of the owner
  *   access=permissive --> Publicly available until further notice
  *   access=no --> Unavailable

and add validation:

if access=customers plesase add opening_hours
if access=yes not allowed to add opening_hours and opening_hours = 24/7 can be added automatically

If someone outside a normal swimming pool is dying, nobody will hesitate to take the AED inside the swimming pool outside. Nor should one.

Of course, but the access time to the device increases. The user of the map should know this. In addition, many such facilities are not available outside working hours.

**Comment by wzbartczak** [comment]
>Did you switch private and permit? Surely, if you need the permit of the owner it’s access=permit?

That is not what I meant, Sorry. 
* private - It is not available to the public, such as in an office or workplace. If there are problems, it will be available, but it will take more time.
* permit - is available in a public area, i.e. it is accessible but requires the consent of the owner. 

The website and app will now be undergoing testing at one of the Polish emergency centres, they may still have additional comments so it is worth establishing this and unifying the wiki.

**Comment by amadeous** [comment]
I recently used OpenAEDMap and experienced the same confusion mentioned here. Could I invite you to move forward with this issue?

Something should be done. Even if the precise details are still undefined, changes to prevent this confusion could be made. Iterations could later be performed to refine the details of access, based on user's input to the previous changes

**Comment by rbovard** [comment]
Thanks for reviving the subject @amadeous

As I [suggested before](https://github.com/openstreetmap-polska/openaedmap-frontend/issues/113#issuecomment-1426029370), a quick change that can be made to avoid this confusion is to rename `Accessible during work hours` to its real meaning: `Accessible only for customers` ([source](https://wiki.openstreetmap.org/wiki/Tag:access%3Dcustomers)).
